Frameworks/Foundation/NSUserDefaults.mm

```diff
@@ -119,16 +132,44 @@ - (void)setPersistentDomain:(id)domain forName:(NSString*)name {
  @Status Interoperable
 */
 - (BOOL)synchronize {
-    return CFPreferencesAppSynchronize(kCFPreferencesCurrentApplication);
+    _cacheLock.lock();
```

Comment:
Fortunately, `std::lock_guard` locks a scope, and you could introduce a scope for this.

---
In reply to: [70674087](https://github.com/Microsoft/WinObjC/pull/612#discussion_r70674087) [](ancestors = 70674087,70540876)